### PR TITLE
Implement perfdash support for ContainerRestarts measurement.

### PR DESCRIPTION
**Pull request by mborsz**
/kind feature

/assign @jprzychodzen

### CODE REVIEW

**Comment by mborsz** [comment]
/hold

It's quite to read. Let me change it slightly

**Comment by mborsz** [comment]
/hold cancel

I changed the code to present all containers on a single graph. The reasoning is that it should be 0 for most of the containers and having one view makes it easier to find restarting components.

File: perfdash/parser.go [hunk 1]
```diff
@@ -52,6 +52,30 @@ func createRequestCountData(data *perftype.DataItem) error {
...
+
+func parseContainerRestarts(data []byte, buildNumber int, testResult *BuildData) {
+	build := fmt.Sprintf("%d", buildNumber)
+	restarts := make([]ContainerRestartsInfo, 0)
+	if err := json.Unmarshal(data, &restarts); err != nil {
+		klog.Errorf("error parsing JSON in build %d: %v %s", buildNumber, err, string(data))
+		return
+	}
+
+	restartsbyContainer := make(map[string]float64)
```
Comment (jprzychodzen) [inline perfdash/parser.go:70]: nit: s/restartsbyContainer/restartsByContainer

File: perfdash/parser.go [hunk 2]
```diff
@@ -52,6 +52,30 @@ func createRequestCountData(data *perftype.DataItem) error {
...
+	for _, restart := range restarts {
+		restartsbyContainer[restart.Container] = restartsbyContainer[restart.Container] + float64(restart.RestartCount)
```
Comment (jprzychodzen) [inline perfdash/parser.go:72]: I'm not sure that 'Container Name' is the best key here. Actually 'ResourceUsage' has a nice pattern of 'simplified(Pod)/Container'. Maybe this is something that we can follow?

What do you think about 
`key = Namespace + "/" + RemoveDisambiguationInfixes(Pod) + "/" + Container`?

File: perfdash/parser.go (hunk 1 above)
Comment (mborsz) [reply perfdash/parser.go:70]: Looks like both parts (restartsbyContainer and restartsbyContainer) are the same?

image

File: perfdash/parser.go (hunk 2 above)
Comment (mborsz) [reply perfdash/parser.go:72]: I was thinking about including pod name in the key, but the problem is that for static pods (which are a very important use case for this) it includes hostname, e.g. fluentbit-gke-something-something-something and the something are changing from test run to test run.

RemoveDisambiguationInfixes will not strip hostnames which is necessary to make it useful for pods running as static pods , do you want to change it to strip hostname part?

How to do that cleanly? Strip everything after "-gke"? It sounds error-prone.

Personally I feel like container name is good enough and still quite clean. If we better key, we can change it later.

File: perfdash/parser.go (hunk 1 above)
Comment (jprzychodzen) [reply perfdash/parser.go:70]: There is capital **B**

File: perfdash/parser.go (hunk 2 above)
Comment (jprzychodzen) [reply perfdash/parser.go:72]: FTR: We could configure this 'cloud provider specific' regex as flag for Perfdash. But it seems like an overblown approach.

LGTM here, let's handle it the way it is. If we will need better naming, we can figure out this as needed.

File: perfdash/parser.go (hunk 1 above)
Comment (mborsz) [reply perfdash/parser.go]: :D thanks!

Done

**Pull request merged**